src/Microsoft.ML.Data/Evaluators/MulticlassClassificationEvaluator.cs

```diff
@@ -334,23 +351,34 @@ public void Update(int[] indices, double loglossCurr, int label, float weight)
                     if (label < _numClasses)
                         _totalPerClassLogLoss[label] += loglossCurr * weight;
 
-                    if (assigned == label)
+                    _seenRanks[seenRank]++;
```

Comment:
To handle the weight, should this be:
```suggestion
                    _seenRanks[seenRank] += weight;
```

We may need additional handling later to get the total weight of correct predictions. I think this can be pulled from the confusion table.

Reply:
Looks like the additional handling is taken care of in the `_numInstances += weight` line. 

But we still need the `+= weight` here.

Reply:
I'll update this.

Reply:
Updated